src/librustc_codegen_ssa/lib.rs

```diff
@@ -117,6 +116,19 @@ bitflags::bitflags! {
     }
 }
 
+#[derive(Copy, Clone, PartialEq, Eq, PartialOrd, Ord, Hash)]
```

Comment:
```suggestion
#[derive(Copy, Clone, PartialEq, Eq, PartialOrd, Ord, Hash, RustcDecodable, RustcEncodable)]
```

Reply:
IMO adding `RustcDecodable` and `RustcEncodable` is beyond the scope of this PR. I prefer to leave it to #67195, where `RustcDecodable` and `RustcEncodable` are added to some other structs and enums.